src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -266,7 +266,7 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr curlPtr)
                 catch (Exception exc)
                 {
                     EventSourceTrace("Unexpected exception: {0}", exc, easy: easy);
-                    easy.FailRequest(exc);
+                    easy.FailRequest(CreateHttpRequestException(exc));
```

Comment:
The value is app-compat and behavior compat.  Do we expect a developer to just use .InnerException to find out what they want.  The first InnerException will be different between the stacks.  Or is the best practice advise to tell them to use GetBaseException() to find out the root cause.  Either way, we are creating a difference.

IMHO I think the design pattern of HttpClient was that HttpRequestException was the top-level and then inside that was the handler-specfic exception which might have further inner exceptions such as i/o exceptions from the network etc.